File: test/src/ctypes_type_test.c
// Copyright (C) 2008-2010 by Yung-Yu Chen.  See LICENSE.txt for terms of
// usage.
#include "ctypes.h"
void ctypes_type_test(record *par_t, record *ret_t) {
	double *locarr;
	int it;
	ret_t->idx = -par_t->idx;
	for (it=0; it<10; it++) {
		ret_t->arr[it] = -par_t->arr[it];
	};
	locarr = par_t->arr_ptr;
	for (it=0; it<10; it++) {
		locarr[it] += 200.0;
	};
	return;
};
// vim: set ts=4 et:

File: obselete/testsrc/ctypes_type_test.f90
subroutine ctypes_type_test(par_t, ret_t)
use iso_c_binding
implicit none
include 'ctypes.inc'
! arguments.
type(record), intent(in) :: par_t
type(record), intent(out) :: ret_t
! locals.
real*8, pointer, dimension(:) :: locarr
real*8, allocatable, dimension(:) :: testarr
ret_t%idx = -par_t%idx
ret_t%arr(:) = -par_t%arr(:)
call c_f_pointer(par_t%arr_ptr, locarr, [par_t%idx])
locarr(:) = locarr(:) + 200.d0
end subroutine ctypes_type_test
